### PR TITLE
DDF-4994 Remove Embedded Solr Provider

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Removes the no longer required EmbeddedSolrProvider.

#### Who is reviewing it? 
@lamhuy 
@rzwiefel 

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@ahoffer
@bdeining 
@paouelle
@pklinef

#### How should this be tested?
Full build with tests.
Unzip and Run and verify basic functionality works as expected (Create workspace, etc)

#### Any background context you want to provide?
The Embedded Solr Provider is no longer required by any code, so removing this to clean up baseline.

#### What are the relevant tickets?
Fixes: #4994 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by troymohl** [comment]
build now

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/ConfigurationStore.java [hunk 1]
```diff
@@ -42,7 +43,7 @@
    *
    * <p>To circumvent the bug that this one place creates when it starts re-configuring the store to
    * use an in-memory Solr server, we were forced to make this constructor public. The {@link
-   * EmbeddedSolrFactory}Solr factory class was properly modified to rely on the passed in
+   * SolrClientFactory}Solr factory class was properly modified to rely on the passed in
```
Comment (paouelle) [inline platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/ConfigurationStore.java]: ✏️ This statement might no longer be true. It was not the SolrClientFactory but really the embedded one as originally stated.

**Comment by paouelle** [comment]
@brjeter How will this affect the upgrade logic you are developing?

**Comment by brjeter** [comment]
@paouelle it shouldn't affect it since the system has been using Solr as a separate process for a long time now.

**Comment by paouelle** [comment]
@brjeter we might want to add a check as part of the upgrade to warn the user if this is detected

**Comment by pklinef** [comment]
@paouelle I think the `CachingFederationStrategy` and `SolrCache` used to have an option to create an in memory Solr server to support a feature for the Standard UI.  Looks like that was properly removed with the Standard UI.

**Comment by lamhuy** [comment]
look like HttpSolrClientFactory is also deprecated. i thought we are still using it....
if it is truly deprecated,  so should this be clean up in this PR as well? 
https://github.com/codice/ddf/blob/master/platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java#L47

**Comment by paouelle** [comment]
@lamhuy What is deprecated is its direct use. You should always go through the SolrClientFactory.

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/ConfigurationStore.java (hunk 1 above)
Comment (troymohl) [reply platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/ConfigurationStore.java]: So is that not necessary anymore now that the embeddedsolrprovider has been removed? Can I just remove this constructor?

**Comment by troymohl** [comment]
I have updated with additional removed code that is no longer needed. @pklinef provided some additional dependencies that could be removed/cleaned up.

**Comment by troymohl** [comment]
build now

**Comment by troymohl** [comment]
build now

**Comment by willwill96** [comment]
### Hero Successful
1. Quickbuilt and unzipped
2. Successfully performed CRUD operations on workspaces, search forms, and records

**Comment by troymohl** [comment]
build now

**Comment by kcwire** [comment]
build now

**Comment by troymohl** [comment]
build now

**Pull request merged**